### PR TITLE
Add support for running second window without arguments

### PR DESCRIPTION
**Pull request by dreisel**
Made to support second window when running without ipc args

### CODE REVIEW

**Comment by asticode** [comment]
I'm having trouble seeing what's your use case here?

**Comment by dreisel** [comment]
if the program runs without the levant arguments, the code reaches the client.js code and crushes.

this behavior will occurs in the flow when the user clicks the default jumplist and windows will trigger the electron.exe with bo arguments.

I'm now working on patching the bundler to modify the default electron app to always open the astielectron app even if no arguments are given, and by that he will call the astillectron app without the client args, so I want to notify the main window on the click of the item.

this won't solve the issue if someone clicks "pin to taskbar" cause then the app will stick to the taskbar where there is no other running instance, but its a start and a more common case

**Comment by asticode** [comment]
In that case I'd rather have `if (process.argv.length > 3 && process.argv[3] === "true") {` since using a single instance is disabled by default

**Comment by dreisel** [comment]
but running without arguments won't enter the condition to quit the app in the case of no args at all.

**Comment by dreisel** [comment]
its a special cases that causes the app to "saftly close" instead of crashing.

I can add a code that calls app.quit(); in the case that there is no second window.

WDYT?

File: main.js [hunk 1]
```diff
@@ -3,6 +3,13 @@
 const { app } = require("electron");
 const { start, lastWindow } = require("./index");
 
+// edge case when the program is launched without arguments
+if (process.argv.length == 1) {
+  app.requestSingleInstanceLock();
```
Comment (asticode) [inline main.js:8]: Why don't you check the lock here ?

**Review by asticode — CHANGES_REQUESTED** [review]

File: main.js (hunk 1 above)
Comment (dreisel) [reply main.js:8]: cause there is nothing to do with the lock,
it only signals to the other window (if subscribed to the event).
then quitting the app cause there is no way the app can run without the host and port args configs

Comment (asticode) [reply main.js:8]: ok makes sense

**Comment by asticode** [comment]
FYI I've bumped GO code as well and created new tags for concerned projects

**Pull request merged**